### PR TITLE
Configurable and improved authentication and authorization

### PR DESCRIPTION
**Pull request by josvandervelde**
1. Use a public client for swagger
2. Perform a back-end authorization request (instead of just decoding the token)
3. Make authorization parameters configurable through environment variables, so that we can use EGI checkin easily

### CODE REVIEW

File: src/.env [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# DO NOT ADD SENSITIVE VALUES HERE!
+# In the end, this file should not be checked in into git.
+# TODO: refactor configuration (https://github.com/aiondemand/AIOD-rest-api/issues/82)
+
+# Authentication
```
Comment (josvandervelde) [inline src/.env:5]: For the reviewers: 

EGI checkin now works :) see https://test.openml.org/demo/docs#/.
I configured this using 
`KEYCLOAK_SERVER_URL=https://aai-demo.egi.eu/auth/
KEYCLOAK_REALM=egi
KEYCLOAK_CLIENT_ID=f681e611-1a8a-47c6-8c2d-c875cc85c734
KEYCLOAK_CLIENT_ID_SWAGGER=aiod-api-swagger
KEYCLOAK_CLIENT_SECRET=[SECRET]
KEYCLOAK_OPENID_CONNECT_URL=https://aai-demo.egi.eu/auth/realms/egi/.well-known/openid-configuration
KEYCLOAK_SCOPES="openid profile email eduperson_entitlement voperson_id"
KEYCLOAK_ROLE=ai4europe-editor
`

**Review by arejula27 — COMMENTED** [review]
I see it ok

**Review by arejula27 — APPROVED** [review]

**Comment by josvandervelde** [comment]
I made a small update. I was afraid that people will check in the `.env` file, so I removed it again from git.

**Review by arejula27 — APPROVED** [review]
Perfect

**Pull request merged**